.tekton/wait-for-image-task.yaml

```diff
@@ -4,28 +4,35 @@ metadata:
   name: wait-for-image
   namespace: rh-acs-tenant
 spec:
-  description: Waits until the specified image is found.
+  description: Waits until the specified image is found and returns its digest.
   params:
   - name: IMAGE
     description: Image reference.
     type: string
-  workspaces:
```

Comment:
For clarification: Is it correct to assume that this secret is not required, because the "global Secret" in Konflux is attached to the service account running our pipelines and that will be used to authenticate calls to quay.io?

Reply:
It's a good question why does it work. I tried flipping the coin and [asked](https://redhat-internal.slack.com/archives/C073B14UE10/p1729598600315119) @mclasmeier whether he observed it _not_ working without the secret but we can only get an answer after Moritz is back.

Meanwhile, I suppose, I need to roll up sleeves and investigate myself. There's something relevant I encountered [here](https://github.com/konflux-ci/oras-container/blob/main/hack/select-oci-auth.sh) so will use that as a starting point.

Reply:
It wasn't hard.

1. Tekton produces `~/.docker/config.json` file where it collects secrets of `kubernetes.io/dockerconfigjson` type (and some others), see <https://tekton.dev/docs/pipelines/auth/#overview>. Our Quay rw secret has `type: kubernetes.io/dockerconfigjson`.
2. Skopeo respects `~/.docker/config.json` for references starting with `docker://`, see [here](https://github.com/containers/skopeo/blob/3161b9faf8db46f27c2527682111208f82de34ab/docs/skopeo.1.md?plain=1#L35-L36). This reference format is what we use.

Therefore, all things work magically.

An approach with explicit pull secret that Moritz used should have also worked according to `skopeo copy` [description](https://github.com/containers/skopeo/blob/3161b9faf8db46f27c2527682111208f82de34ab/docs/skopeo-copy.1.md?plain=1#L40-L41).

Therefore, it becomes a matter of choice: explicit or automagic. Since we cannot disable this mechanism (it seems to be an [infra-level setting](https://tekton.dev/docs/pipelines/auth/#disabling-tektons-built-in-auth)) and other OOTB tasks rely on this (otherwise buildah* tasks would not be able to push images to quay.io/rhacs-eng/* without extra configuration), I suggest we also rely on "automagic".

Should I have brought this up in some earlier PR review? Yes, but I missed a review train for <https://github.com/stackrox/stackrox/pull/12797>.

Reply:
Thanks for investigating, it works as I had in mind. Let's keep it automagic, consistent with other tasks.